api/lua/examples/add-function-get-arg-string.rs

```diff
@@ -0,0 +1,68 @@
+#![no_std]
+extern crate alloc;
+
+#[macro_use]
+extern crate sys;
+extern crate playdate_lua as lua;
+
+use core::ffi::c_int;
+use core::ptr::NonNull;
+
+use lua::Lua;
+use sys::EventLoopCtrl;
+use sys::ffi::*;
+use system::System;
+use system::event::SystemEventExt as _;
+use system::update::UpdateCtrl;
+
+
+/// Entry point, event handler
+#[no_mangle]
+fn event_handler(_api: NonNull<PlaydateAPI>, event: PDSystemEvent, _: u32) -> EventLoopCtrl {
+	// We need to set our update callback in the InitLua handler instead of Init.
+	// https://devforum.play.date/t/lua-c-minimal-example/4354/5
+	//
+	// Just for this example, ignore all other events.
+	if event != PDSystemEvent::InitLua {
+		return EventLoopCtrl::Continue;
+	}
+
+	// Set update callback
+	System::Default().set_update_callback_static(Some(on_update), ());
+
+	// Add a function that we depend on and call in main.lua
+	Lua::Default().add_function(Some(log_to_console_from_main_dot_lua), "example.logToConsole")
```

Comment:
The `Some` here is a result of using `lua_CFunction`.

We could remove `Option` from the exposed API but I suppose someone might want to unset this by providing `None` for some reason.

Reply:
I have in plan to describe Callback abstraction with some traits like `IntoCallback` that impl for `Option<Callback>` for example. So with this idea you will able to call `set_update_callback_static` with `None`, `Some(Fn)` as well as just fn. That will be cool. But currently that is just an idea.

Reply:
LGTM 👍🏻, `Some` there is ok.